snow/consensus/snowman/topological.go

```diff
@@ -287,48 +297,9 @@ func (ts *Topological) RecordPoll(ctx context.Context, voteBag bag.Bag[ids.ID])
 	}
 
 	// Runtime = |live set| ; Space = Constant
-	preferred, err := ts.vote(ctx, voteStack)
-	if err != nil {
-		return err
-	}
-
-	// If the set of preferred IDs already contains the preference, then the
-	// tail is guaranteed to already be set correctly. This is because the value
-	// returned from vote reports the next preferred block after the last
-	// preferred block that was voted for. If this block was previously
-	// preferred, then we know that following the preferences down the chain
-	// will return the current tail.
-	if ts.preferredIDs.Contains(preferred) {
-		return nil
-	}
```

Comment:
This PR removes this optimization. Should look into how important it was.

Reply:
Is this because it is incompatible with these changes or for some other reason?

Reply:
It wasn't really compatible - because we may still want to update our preference due to something passing verification that previously failed... When we change the code to evict invalid blocks - I think this optimization makes sense again.